src/utils/wallet.js

```diff
@@ -153,10 +154,31 @@ export class Wallet {
         if (this.isEmpty()) {
             throw new Error('No account.')
         }
+
+        const [ fromChain, fromDb ] = await Promise.all([
+            this.getAccount(this.accountId).state(),
+            sendJson('POST', `${ACCOUNT_HELPER_URL}/account/recoveryMethods`, {
+                accountId: this.accountId,
+                ...(await this.signatureFor(this.accountId))
+            })
+        ]);
+
+        const recoveryMethods = {}
+        if (fromDb.email) {
+            recoveryMethods.email = { data: fromDb.email, timeStamp: fromDb.emailAddedAt }
+        }
+        if (fromDb.phoneNumber) {
+            recoveryMethods.phone = { data: fromDb.phoneNumber, timeStamp: fromDb.phoneAddedAt }
+        }
+        if (fromDb.phraseAddedAt) {
+            recoveryMethods.phrase = { timeStamp: fromDb.phraseAddedAt }
```

Comment:
@chadoh @Patrick1904  I think we have pretty big problem that none of these have info on what are associated public keys, so we have no idea which key to remove if you want to replace/remove given recovery method.

Reply:
My original plan to address it was to add `meta` to the access keys which is loaded from elsewhere:
https://github.com/nearprotocol/near-wallet/blob/b7bbfbeef97c64d668cfe07385b377a6d0bac6e3/src/utils/wallet.js#L170

Reply:
Oh, great point! Should we be storing that information in the `near-contract-helper` database?

Reply:
@chadoh I think long term we should be storing this information on chain using open web contract from @evgenykuzyakov. We can store it in near-contract-helper short term if it's easier to get working.

Reply:
Stored in near-contract-helper now (pending merge of nearprotocol/near-contract-helper#88)